### ISSUE TITLE
Constant Specification Update 2018

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/language#61]
This is the feature proposal for request https://github.com/dart-lang/language/issues/54. It improved the support for constant expressions, for example the ability to explicitly cast a value in a const expression:

```dart
class MyClass {
  final String value;
  const MyClass(Object o) : value = (o is! String) ? "$o" : (o as String);
}
```

The feature specification is:
https://github.com/dart-lang/language/blob/master/accepted/2.5/constant-update-2018/feature-specification.md

**Comment by jodinathan** [comment]
with this will it be possible to use functions in metadata?
eg:
```
class Foo {
  @Bar(() => 'foobar')
  String myProp;
}
```

**Comment by lrhn** [comment]
@jodinathan 
No, there is no change to function literals in the listed changes, only small tweaks to include recently added operators and make the constant evaluation better match Dart 2 type semantics.

You have to do:
```
String kFooBar() => 'fooBar';
class Foo {
  @Bar(kFooBar)
  String myProp;
}
```
to tell the compiler that the function is a constant.

**Comment by jodinathan** [comment]
@lrhn that does not work either.

**Comment by mit-mit** [comment]
This shipped in Dart 2.5
